File: Source/JapaneseLanguageTools.Contracts/Models/Requests/GenerateCharacterExerciseBatchRequest.Model.cs
using System.Collections.Generic;

using JapaneseLanguageTools.Contracts.Enumerations;

namespace JapaneseLanguageTools.Contracts.Models.Requests;

public class GenerateCharacterExerciseBatchRequestModel
{
    public int Size { get; set; }

    public required CharacterExerciseProfileModel? UseCharacterExerciseProfile { get; set; }

    public IList<int> UseCharacterGroupIds { get; set; } = [];

    public CharacterTypes UseCharacterTypes { get; set; }
}

File: Source/JapaneseLanguageTools.ReactSpa/source/models/requests/GenerateCharacterExerciseBatchRequest.Model.ts
import CharacterTypes from "@/enumerations/CharacterTypes";
import CharacterExerciseProfileModel from "@/models/CharacterExerciseProfile.Model";

export default interface GenerateCharacterExerciseBatchRequestModel {
  size: number;
  useCharacterExerciseProfile?: CharacterExerciseProfileModel;
  useCharacterGroupIds: number[];
  useCharacterTypes: CharacterTypes;
}
